### PR TITLE
Migrate build to Maven and add Travis

### PR DESCRIPTION
**Pull request by vlsi**
I've migrated build to Maven, and added Travis configuration.
I've removed `doc/api` and `lib`.  `lib/jgraphx.jar` can be shared via Maven Central.
javadoc could be served either via Maven Central (regular javadoc integration in IDEs) or via separate branch.

**Note:** maven coordinates are `com.github.vlsi.mxgraph:*` as I'm testing it via my own account.
**Note:** readme should be improved to account maven build instructions, release instructions, example execution instructions.

You can see Travis CI here: https://travis-ci.org/vlsi/jgraphx (current javadoc fails under java 8)

### CODE REVIEW

**Comment by davidjgraph** [comment]
I appreciate your efforts, however, this seems to basically rewrite our build system just to push to Maven. I'm not overly keen on Maven dictating what we store in Github, regarding removing the JavaDocs and binary. And as for changing the entire directory structure...

**Comment by vlsi** [comment]
> just to push to Maven

Well, the main reasoning was not maven central, but "making the build more developer and IDE friendly".

Maven/Gradle can be imported as a mere "open project file", while ant projects typically require different tweaks.

**Comment by vlsi** [comment]
Here's the reasoning: https://github.com/jgraph/jgraphx/issues/28#issuecomment-223808595

**Comment by davidjgraph** [comment]
Wouldn't just having a correct pom.xml be enough?

**Comment by vlsi** [comment]
It might be, however:
1) If that "pom.xml" would be optional, it would easily broke over time and no one would notice
2) It would implement the same things multiple times (first for ant, then for pom.xml). It would look like a code duplication for little reason
3) Even though maven allows to customize lots of things like folders, file names, etc, however I'm sure it would screw IDEs if used extensively

In other words, I do not think "pom.xml to make IDE happy" is a good way to go.

**Comment by vlsi** [comment]
Don't get me wrong.
I'm not forcing into maven. It is fine if you decline "mavenizaton".

The moment I went over jgraphx I did mean it were so much easier to build/contribute if IDE could pick up the project, etc.


A side benefit of pushing javadocs to central is services like "javadoc viewer": http://javadoc.io/doc/org.postgresql/postgresql/9.4.1211.jre7
